ConfigurationSystem/Agent/RucioSynchronizer.py

```diff
@@ -0,0 +1,359 @@
+
+            result = Operations().getSections('Shares')
+            if result['OK']:
+                for dataLevel in result['Value']:
+                    result = Operations().getOptionsDict('Shares/%s' % dataLevel)
+                    if result['OK']:
+                        rseDict = result['Value']
+                        for rse in rses:
+                            try:
+                                self.log.info('Setting %sShare for %s : %s' % (dataLevel, rse, rseDict.get(rse, 0)))
+                                client.add_rse_attribute(rse, '%sShare' % dataLevel, rseDict.get(rse, 0))
+                            except Exception as err:
+                                self.log.error('Cannot create %sShare for %s' % (dataLevel, rse))
+                    else:
+                        self.log.error('Cannot get SEs : %s' % result['Value'])
+            else:
+                self.log.error('Cannot get shares : %s' % result['Value'])
+
+            # Create the RSE attribute PrimaryDataSE and OccupancyLFN
+            result = gConfig.getValue('Resources/StorageElementGroups/PrimarySEs')
+            result = getStorageElements()
+            if result['OK']:
+                allSEs = result['Value']
+                primarySEs = resolveSEGroup('PrimarySEs', allSEs)
+                self.log.info('Will set primarySEs flag to %s' % str(primarySEs))
+                for rse in rses:
+                    if rse in allSEs:
+                        storage = StorageElement(rse)
+                        OccupancyLFN = storage.options.get('OccupancyLFN')
+                        try:
+                            client.add_rse_attribute(rse, 'OccupancyLFN', OccupancyLFN)
+                        except Exception as err:
+                            self.log.error('Cannot create RSE attribute OccupancyLFN for %s : %s' % (rse, str(err)))
+                    if rse in primarySEs:
+                        try:
+                            client.add_rse_attribute(rse, 'PrimaryDataSE', True)
+                        except Exception as err:
+                            self.log.error('Cannot create RSE attribute PrimaryDataSE for %s : %s' % (rse, str(err)))
+                    else:
+                        try:
+                            client.delete_rse_attribute(rse, 'PrimaryDataSE')
+                        except RSEAttributeNotFound:
+                            pass
+                        except Exception as err:
+                            self.log.error('Cannot remove RSE attribute PrimaryDataSE for %s : %s' % (rse, str(err)))
+
+
+            # Collect the user accounts from Dirac Configuration and create user accounts in Rucio
+            self.log.info("Synchronizing accounts")
+            listAccounts = [str(acc['account']) for acc in client.list_accounts()]
+            listScopes = [str(scope) for scope in client.list_scopes()]
+            accounts = getAllUsers()
```

Comment:
I think there is no need for this to be a variable, it can simply be `for account in getAllUsers():`

Reply:
Done